lib/rubocop/cop/mixin/negative_conditional.rb

```diff
@@ -6,15 +6,20 @@ module Cop
     # Some common code shared between FavorUnlessOverNegatedIf and
     # FavorUntilOverNegatedWhile.
     module NegativeConditional
+      def self.included(mod)
+        mod.def_node_matcher :double_negative?, '(send (send _ :!) :!)'
+      end
+
       def check_negative_conditional(node)
         condition, _body, _rest = *node
 
         # Look at last expression of contents if there are parentheses
         # around condition.
         condition = condition.children.last while condition.type == :begin
-        return unless condition.type == :send
+        return unless condition.type == :send && !double_negative?(condition)
 
         _object, method = *condition
+
         return unless method == :! && !(node.loc.respond_to?(:else) &&
```

Comment:
Ah, yeah. I got confused here. Still, it seems to me we can directly check that we're dealing with single instead of double negation and that would simplify the logic. Right now the structure of the methods seems a bit convoluted.